### DISCUSSION TITLE
0.14.0

### DISCUSSION THREAD

**Discussion by arlyon** [arlyon/async-stripe#194] [Announcements]
Alright! Happy to announce we have a new release of async-stripe. This is a big one with a number of changes behind the scenes to enable our future goals (outlined here: https://github.com/arlyon/async-stripe/discussions/186). But, for now, lets celebrate what we've completed so far.

- **improve test coverage**: we added tarpaulin and started tracking coverage. from now on new contributions will be expected to be tested properly, and we will be adding extensive examples
- **piles of examples**: we have a whole load of shiny examples to look at, along with example-doc linking to put the examples straight alongside the code in docs.rs. very happy with the ergonomics here.
- **make sure the 'unusual params' are implemented properly**: we added `card` to the api! :partying_face: 
- **explore the effects of implementing `Default` across the codebase**: we added default everywhere! #152 
- **allow the openapi generator to box in specific places**: there is a simple extensible heuristic for this! #160 
- **opt-in auto-retry functionality following the stripe retry headers**: the client will retry failed requests :tada: 
- **opt in idempotency**: now you can make the same mistake twice :P 

## New Contributors
* @andrewhalle made their first contribution in https://github.com/arlyon/async-stripe/pull/164
* @SmailBarkouch made their first contribution in https://github.com/arlyon/async-stripe/pull/170
* @lms11 made their first contribution in https://github.com/arlyon/async-stripe/pull/173

**Full Changelog**: https://github.com/arlyon/async-stripe/compare/v0.13.0...v0.14.0

<hr /><em>This discussion was created from the release <a href='https://github.com/arlyon/async-stripe/releases/tag/v0.14.0'>0.14.0</a>.</em>

**Comment by erichCompSci** [comment]
Looking forward to being able to find some time to help contribute this project some more in the coming summer months.  Arlyon, is the branch that you're using to enhance the code public?  Should I work off of that or do you want me to target specific things branched off of main?  If so, what things are you hoping to tackle?  I would like to tackle the code gen, but I don't want to repeat any work you've already done while I've been busy. @arlyon (not sure if this notifies you or not, but I'm ating you anyway.)

**Reply by arlyon** [reply]
Hey! Yes aside from some minor hacking what is on mainline is up to date. I have some WIP branches that I've been trying to make time for, but they should be up there. Looking forward to seeing what you come up with!

**Reply by erichCompSci** [reply]
Okay, it doesn't look like some minor hacking, the code is a lot more clean than when I last looked at it.  It looks like mainline is failing because we need to add code to generated.rs and resources.rs.  After we clean up the interface to generate the missing API's, then we need to tackle that issue too so that we don't have to keep hand writing that code.  Not sure what that's going to look like, as right now we expose only arbitrary portions of the code, but we can figure that out later.

**Reply by arlyon** [reply]
I did some experimentation a while back on this branch https://github.com/arlyon/async-stripe/pull/197 maybe some of the progress there would be interesting to you.

First step is breaking things into crates, which I have done, now I'm trying to work on heuristics for all the imports between crates. The challenge here is trying to figure out cyclic dependencies. I have a feeling we may end up with types that break things.
